test/Microsoft.Dnx.Runtime.Tests/DependencyManagement/FrameworkReferenceResolverFacts.cs

```diff
@@ -0,0 +1,131 @@
+        [InlineData("net46", "System.Printing", @"REFASMSROOT\.NETFramework\v4.6\System.Printing.dll", "4.0.0.0")]
+
+        // System.Core (added in 3.5)
+        [InlineData("net35", "System.Core", @"REFASMSROOT\v3.5\System.Core.dll", "3.5.0.0")]
+        [InlineData("net40", "System.Core", @"REFASMSROOT\.NETFramework\v4.0\System.Core.dll", "4.0.0.0")]
+        [InlineData("net45", "System.Core", @"REFASMSROOT\.NETFramework\v4.5\System.Core.dll", "4.0.0.0")]
+        [InlineData("net451", "System.Core", @"REFASMSROOT\.NETFramework\v4.5.1\System.Core.dll", "4.0.0.0")]
+        [InlineData("net452", "System.Core", @"REFASMSROOT\.NETFramework\v4.5.2\System.Core.dll", "4.0.0.0")]
+        [InlineData("net46", "System.Core", @"REFASMSROOT\.NETFramework\v4.6\System.Core.dll", "4.0.0.0")]
+
+        // Microsoft.CSharp (added in 4.0)
+        [InlineData("net40", "Microsoft.CSharp", @"REFASMSROOT\.NETFramework\v4.0\Microsoft.CSharp.dll", "4.0.0.0")]
+        [InlineData("net45", "Microsoft.CSharp", @"REFASMSROOT\.NETFramework\v4.5\Microsoft.CSharp.dll", "4.0.0.0")]
+        [InlineData("net451", "Microsoft.CSharp", @"REFASMSROOT\.NETFramework\v4.5.1\Microsoft.CSharp.dll", "4.0.0.0")]
+        [InlineData("net452", "Microsoft.CSharp", @"REFASMSROOT\.NETFramework\v4.5.2\Microsoft.CSharp.dll", "4.0.0.0")]
+        [InlineData("net46", "Microsoft.CSharp", @"REFASMSROOT\.NETFramework\v4.6\Microsoft.CSharp.dll", "4.0.0.0")]
+
+        // Microsoft.Build.Engine (in 2.0, but overridden in 3.5)
+        [InlineData("net20", "Microsoft.Build.Engine", @"%WINDIR%\Microsoft.NET\Framework\v2.0.50727\Microsoft.Build.Engine.dll", "2.0.0.0")]
+        [InlineData("net30", "Microsoft.Build.Engine", @"%WINDIR%\Microsoft.NET\Framework\v2.0.50727\Microsoft.Build.Engine.dll", "2.0.0.0")]
+        [InlineData("net35", "Microsoft.Build.Engine", @"REFASMSROOT\v3.5\Microsoft.Build.Engine.dll", "3.5.0.0")]
+
+        public void FrameworkResolverResolvesCorrectPaths(string shortFrameworkName, string assemblyName, string expectedPath, string expectedVersion)
+        {
+            var resolver = new FrameworkReferenceResolver();
+
+            string actualPath;
+            Version actualVersion;
+            Assert.True(resolver.TryGetAssembly(assemblyName, VersionUtility.ParseFrameworkName(shortFrameworkName), out actualPath, out actualVersion));
+            Assert.Equal(Environment.ExpandEnvironmentVariables(expectedPath).Replace("REFASMSROOT", FrameworkReferenceResolver.GetReferenceAssembliesPath()), actualPath);
+
+            // Having this be Version->Version equality caused some problems...
+            Assert.Equal(Version.Parse(expectedVersion).ToString(), actualVersion.ToString());
+        }
+
+        [ConditionalTheory]
+        [FrameworkSkipCondition(RuntimeFrameworks.CLR | RuntimeFrameworks.CoreCLR)]
+        [OSSkipCondition(OperatingSystems.Win2008R2 | OperatingSystems.Win7 | OperatingSystems.Win7And2008R2)]
```

Comment:
`OperatingSystems.Win7And2008R2` already includes the other two. The OSSkip condition is superfluous because of the previous one